pkgs/utils/src/cast.zig

```diff
@@ -0,0 +1,15 @@
+const std = @import("std");
+
+pub fn Cast(To: type, from: anytype) To {
+    var result: To = undefined;
+    inline for (@typeInfo(To).@"struct".fields) |field| {
+        const field_value = switch (@typeInfo(field.type)) {
+            .optional => @field(from, field.name),
```

Comment:
is this what you want to do here? because the way I see it, if the target field isn't an optional and the source field is, then you need to check that the source field isn't `null` and either copy the data if it's there, or return an error/panic otherwise.

But here, what I understand is that you're trying to do the opposite: if the target is an optional, then you assign the value of the source (whether it's the same or not). That should work out of the box without having to do it explicitly, right?

Reply:
see https://zig.fly.dev/p/UROz_wVocn8 for what I mean

Reply:
i needed to do this if target wasn't an optional and source was an optional non null type, to make sure the target type can be properly interpreted in else section. otherwise it was giving compile error